### PR TITLE
Chore 2453/upgrade to java 21

### PR DESCRIPTION
**Pull request by mkimberlin**

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -151,8 +151,8 @@ application {
     mainClass.set("com.objectcomputing.checkins.Application")
 }
 java {
-    sourceCompatibility = JavaVersion.toVersion("17")
-    targetCompatibility = JavaVersion.toVersion("17")
+    sourceCompatibility = JavaVersion.toVersion("21")
```
Comment (mjperry91) [inline server/build.gradle:154]: These settings are also in the web-ui/build.gradle and set to 17. Probably doesn't matter, but might be worth bumping them as well.

**Review by mjperry91 — APPROVED** [review]

File: server/build.gradle (hunk 1 above)
Comment (mkimberlin) [reply server/build.gradle:154]: Good catch. Fixed.

File: docs/getting-started/setup/macos.md [hunk 2]
```diff
@@ -112,9 +112,9 @@ Before running the above command, it's recommended you review the script content
 Then, install the JDK with:
 
 ```shell
-sdk install java 17.0.10-tem
-sdk use java 17.0.10-tem
-sdk default java 17.0.10-tem
+sdk install java 21.0.03-tem
+sdk use java 21.0.03-tem
+sdk default java 21.0.03-tem
```
Comment (vhscom) [inline docs/getting-started/setup/macos.md:117]: There is a `21.0.3-tem` version but no `.03`. There's also an umbrella `21-tem` option for SDKMAN.

**Review by vhscom — APPROVED** [review]
Looks good. Just one small gotcha in the docs.

**Pull request merged**